peak_finder/line_group.py

```diff
@@ -230,9 +230,50 @@ def get_anomaly_attributes(self, line_data_subset: list[LineData]) -> tuple:
             full_peak_values,
         )
 
-    def group_n_groups(  # pylint: disable=R0914
-        self, groups: list[AnomalyGroup]
-    ) -> list[AnomalyGroup]:
+    def accumulate_groups(self, path, neighbourhood) -> list[np.ndarray]:
+        """
+        Accumulate neighbouring groups.
+        """
+        if len(np.unique(path)) == self.n_groups:
+            return [path]
+
+        next_neighbours = np.where(path[-1, 1] == neighbourhood[:, 0])[0]
+
+        if len(next_neighbours) == 0:
+            return [path]
+
+        branches = []
+        for next_neighbour in next_neighbours:
+            current = np.vstack([path, neighbourhood[next_neighbour]])
+            paths = self.accumulate_groups(current, neighbourhood)
+
+            for branch in paths:
+                branches.append(branch)
+
+        return branches
+
+    def find_neighbour_groups(self, groups: list[AnomalyGroup]) -> list:
```

Comment:
unit test

Reply:
I haven't done a full unit test for it as it is pretty involved. Need to create line data, then anomalies. It sort of baked in the peak finder tests though...

Reply:
Did do a big cleanup of the AnomalyGroup class though

Reply:
sure